text/0000-build-std.md

```diff
@@ -0,0 +1,2390 @@
+    target name so JSON targets don't end up enabling `restricted_std`
+    - Need to work out what guarantees are desired
+  - Should std-supported be a property of the target, not `build.rs`?
+  - Idea behind the current approach:
+    - Slightly modified target specs should continue to work and permit usage of the
+      standard library
+    - Different target specs that are very different from existing targets ought not
+      permit usage of the standard library
+  - [rust#71009] needs to be resolved first
+    - Aims to de-stabilise target specifications, making it prohibited to pass one to
+      the compiler
+    - cc [wg-cargo-std-aware#6]
+  - If target-spec-json is unstable then their behaviour with build-std is also unstable
+- **[wg-cargo-std-aware#92]: Consider flag to mark unstable targets**, [madsmtm], Feb 2024
+  - Tier three targets are required to use nightly because of build-std
+    - Stable build-std effectively stabilises these targets, so users would stop using
+      nightly and get fixes less frequently
+  - Is a notion of unstable or incomplete targets desirable?
+- **[wg-cargo-std-aware#95]: Use `build-std=core` on a custom target
+  automatically**, [nazar-pc], Mar 2024
+  - Closed as duplicate of [wg-cargo-std-aware#43]
+
+### Implementation
+[implementation]: #implementation
+
+These issues include bug reports for the current unstable implementation of
+build-std, as well as unresolved questions for the implementation, both from
+[wg-cargo-std-aware]. In addition, this section will list the pull requests in
+[rust-lang/cargo] and [rust-lang/rust] that directly contributed to the
+implementation.
+
+- **[cargo#7216]: Basic standard library support.**, [ehuss], Aug 2019
+  - Initial implementation of `-Zbuild-std`
+  - Constructed a synthetic workspace for the standard library and used
+    `--extern` to provide the new dependencies to rustc.
+  - Required `--target` to be passed so that the host sysroot was used for build
+    scripts and procedural macros.
+- **[cargo#7336]: Add `alloc` and `proc_macro` to libstd crates**, [alexcrichton], Sep 2019
+  - Ensures `alloc` and `proc_macro` are also built when `std` is
+- **[wg-cargo-std-aware#25]: Remove requirement for `--target`**, [ehuss], Sep 2019
+  - `--target` must be specified to require Cargo run in cross-compilation mode (as
+    the `proc_macro` crate needs the host sysroot)
+  - Fixed in [cargo#14317]
+- **[wg-cargo-std-aware#27]: Possibly publish a synthetic `Cargo.toml`**, [ehuss], Sep 2019
+  - build-std initially had to create a false `Cargo.toml` for the standard library as it
+    was part of a workspace with the rest of rust-lang/rust
+  - Fixed by [cargo#14358] (after [rust#128534])
+- **[wg-cargo-std-aware#28]: Fixup some crate flags**, [ehuss], Sep 2019
+  - Some crates require special compiler flags which are scattered throughout bootstrap
+    - e.g. `compiler_builtins` uses `debug-assertions=no`, `codegen-units=1`, `panic=abort`
```

Comment:
`codegen-units=1` is true but unfortunately the situation is more complex:

https://github.com/rust-lang/rust/blob/0d6ab209c525e276cbe7544cbd39a3c3619b6b18/library/Cargo.toml#L17-L27